eland/operations.py

```diff
@@ -233,11 +236,18 @@ def _metric_aggs(
         if numeric_only:
             # Consider if field is Int/Float/Bool
             fields = [field for field in fields if (field.is_numeric or field.is_bool)]
+        elif not numeric_only and (pd_aggs == ["quantile"]):
```

Comment:
Can we check field compatibility within `field.es_agg_compatible()` instead of here?

Reply:
No we cannot, because we aren't converting them to `es_aggs` at that stage. Post filtering the `numeric_only` fields, we usually convert into `es_aggs` from `pd_aggs`. 
:)